gems/smithy/lib/smithy/templates/client/types.erb

```diff
@@ -16,6 +17,27 @@ module <%= namespace %>
     ) do
       include Smithy::Client::Structure
     end
+<% elsif type.type == 'union' -%>
```

Comment:
I think I forgot to cover this in the Doc but is there any changes that needs to make to `UnionShape` itself? Should it have type references to the subclasses from the POV of Union Shape Schema.

Reply:
Good question - it does have a reference to the top level class which would let you get all of the member classes (but does require runtime inspection of the object space)... so maybe we should have a map of member to type?

Reply:
Might be good to implement and useful for customers to access a subclass easily. I want to say the UnionShape in V3 had a way to access to  its member subclasses.